### ISSUE TITLE
issue_service edit method cast error

### ISSUE THREAD

**Issue by houko** [SpinlockLabs/github.dart#179]
file `issue_service.dart`

```
  /// Edit an issue.
  ///
  /// API docs: https://developer.github.com/v3/issues/#edit-an-issue
  Future<Issue> edit(RepositorySlug slug, int issueNumber, IssueRequest issue) {
    return _github
        .request("PATCH", '/repos/${slug.fullName}/issues/$issueNumber',
            body: issue.toJSON())
        .then((response) {
      return Issue.fromJSON(jsonDecode(response.body) as Map<String, dynamic>)
          as Future<Issue>;
    });
  }
```
the last line as Future<issue> is strange, Issue.fromJSON return value is not a Future, and this method also not an async method. why as Future?

the log

```
I/flutter (13477): type 'Issue' is not a subtype of type 'Future<Issue>' in type cast
I/flutter (13477): #0      IssuesService.edit.<anonymous closure> (package:github/src/common/issues_service.dart:133:11)
```

**Comment by robrbecker** [comment]
Thanks for spotting this issue. It looks like this method should be marked async that it would correctly return a Future<Issue>. Putting up a PR now.

**Comment by robrbecker** [comment]
Will be released in the next release: 5.4.0
